hasher-matcher-actioner/hmalib/lambdas/hashing.py

```diff
@@ -71,30 +78,46 @@ def lambda_handler(event, context):
         if message.get("Event") == "s3:TestEvent":
             continue
 
+        media_to_process: t.List[t.Union[S3ImageSubmission, URLSubmissionMessage]] = []
+
         if URLSubmissionMessage.could_be(message):
-            submission_message = URLSubmissionMessage.from_sqs_message(message)
+            media_to_process.append(URLSubmissionMessage.from_sqs_message(message))
+        elif S3ImageSubmissionBatchMessage.could_be(message):
+            # S3 submissions can only be images for now.
+            media_to_process.extend(
+                S3ImageSubmissionBatchMessage.from_sqs_message(
+                    message, image_prefix=IMAGE_PREFIX
+                ).image_submissions
+            )
         else:
             logger.warn(f"Unprocessable Message: {message}")
 
-        content_type = get_content_type_for_name(submission_message.content_type)
-        if not hasher.supports(content_type):
-            logger.warn(f"Unprocessable content type: {content_type}")
-            continue
-
-        with metrics.timer(metrics.names.hasher.download_file):
-            bytes_: bytes = URLContentSource().get_bytes(submission_message.url)
-
-        for signal in hasher.get_hashes(
-            submission_message.content_id, content_type, bytes_
-        ):
-            hash_record = PipelineHashRecord(
-                content_id=submission_message.content_id,
-                signal_type=signal.signal_type,
-                content_hash=signal.signal_value,
-                updated_at=datetime.datetime.now(),
-            )
-
-            hasher.write_hash_record(records_table, hash_record)
-            hasher.publish_hash_message(sqs_client, hash_record)
+        for media in media_to_process:
+            if not hasher.supports(media.content_type):
+                logger.warn(f"Unprocessable content type: {media.content_type}")
```

Comment:
I think we want this to be an error to it'll show up red on the dashboard? (unless doing so results in batch retries)

Reply:
It does result in retries.